### PR TITLE
Add io.github.petitlapin.Li-ri

### PR DESCRIPTION
**Pull request by petitlapin**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

I still have these 4 warnings at linting that I haven't managed to tackle:
{
    "errors": [
        "appstream-missing-appinfo-file",
        "appstream-missing-appinfo",
        "appstream-metainfo-missing",
        "appstream-missing-icon-file"
    ]
}

### CODE REVIEW

**Comment by petitlapin** [comment]
Also, this version should probably replace https://github.com/flathub/net.sourceforge.Ri-li/blob/master/net.sourceforge.Ri-li.json.

Ri-li has been abandoned since 2007, this version takes the latest code, and update SDL1.2 to SDL2 among other updates.

Is there a way to tell one repo should replace another one?

File: org.liri.liri.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "command": "Li-ri",
+  "finish-args": [
+    "--socket=pulseaudio",
+    "--share=ipc",
+    "--socket=fallback-x11",
+    "--socket=wayland",
+    "--device=dri"
+  ],
+  "modules": [
+    "shared-modules/SDL2/SDL2-with-libdecor.json",
```
Comment (bbhtt) [inline org.liri.liri.json:15]:
```suggestion
```

It's in 23.08 runtime.

File: org.liri.liri.json [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        "--enable-music-mod-modplug=yes"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://github.com/libsdl-org/SDL_mixer/releases/download/release-2.6.2/SDL2_mixer-2.6.2.tar.gz",
+          "sha256": "8cdea810366decba3c33d32b8071bccd1c309b2499a54946d92b48e6922aa371"
+        }
+      ]
+    },
```
Comment (bbhtt) [inline org.liri.liri.json:38]:
```suggestion
```

It's in 23.08 runtime https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/05b660171170dbd3fd80d27ac578c36ca3e9eb21/elements/platform.bst#L157

**Comment by bbhtt** [comment]
```

"errors": [
"appstream-missing-appinfo-file",
"appstream-missing-appinfo",
"appstream-metainfo-missing",
"appstream-missing-icon-file"
]
}
```

You need to install appdata, icon, desktop file at the proper locations. Looks like your cmakelists.txt isn't handling appdata

File: org.liri.liri.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
+{
+  "id": "org.liri.liri",
```
Comment (bbhtt) [inline org.liri.liri.json:2]: liri.org redirects to https://www.domaination.com/lander/liri.org?domain=liri.org&redirect=ono-redirect

Did you see the PR checklist?

Comment (petitlapin) [reply org.liri.liri.json:2]: I though it was to know if I controlled the github repository, not the domain.
No, I don't own (and don't plan) the liri.org domain. What do I need to update?

File: org.liri.liri.json (hunk 2 above)
Comment (petitlapin) [reply org.liri.liri.json:37]: It seems the runtime does not compile/link with modplug and sounds are not played for my application if I remove it

File: org.liri.liri.json (hunk 3 above)
Comment (hfiguiere) [reply org.liri.liri.json:2]: Also:

image

So basically you can't use `org.li_ri` either.

Comment (hfiguiere) [reply org.liri.liri.json:2]: (previous comment deleted)

Use something related to your github.

File: org.liri.liri.json (hunk 2 above)
Comment (bbhtt) [reply org.liri.liri.json:37]: Hm you're right, in case keep it.

File: org.liri.liri.json (hunk 3 above)
Comment (hfiguiere) [reply org.liri.liri.json:2]: > I though it was to know if I controlled the github repository, not the domain.

No. It says:

>  The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). 

Is it unclear?

You just said you did NOT intend to have `liri.org`.

Comment (bbhtt) [reply org.liri.liri.json:2]: Eg. `io.github.username.Li-Ri`


> What do I need to update?

ids in appdata, icon, desktop file, if you use the id anywhere in code etc. Then the id here in the manifest and filename.

**Comment by hfiguiere** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by hfiguiere** [comment]
Can't find the appstream file. The name needs to match the app-id.

**Comment by petitlapin** [comment]
I still have an error when running the builder-lint that I don't manage to solve:
```
flatpak run --command=flatpak-builder-lint org.flatpak.Builder repo repo
{
    "errors": [
        "appstream-screenshots-not-mirrored-in-ostree",
        "appstream-missing-icon-file"
    ]
}
```

Looking at the logs, they seem exported:
```
Exporting share/applications/io.github.petitlapin.Li-ri.desktop
Exporting share/icons/hicolor/16x16/apps/io.github.petitlapin.Li-ri.png
Exporting share/icons/hicolor/128x128/apps/io.github.petitlapin.Li-ri.png
Exporting share/icons/hicolor/32x32/apps/io.github.petitlapin.Li-ri.png
Exporting share/icons/hicolor/48x48/apps/io.github.petitlapin.Li-ri.png
Exporting share/icons/hicolor/64x64/apps/io.github.petitlapin.Li-ri.png
Exporting share/appdata/io.github.petitlapin.Li-ri.appdata.xml
```

and the name matches the one in the desktop file: [Icon=io.github.petitlapin.Li-ri](https://github.com/petitlapin/Li-Ri/blob/v3.1.3/io.github.petitlapin.Li-ri.desktop#L8)

**Comment by hfiguiere** [comment]
1. the app-id in the appstream is incorrect https://github.com/petitlapin/Li-Ri/blob/v3.1.3/io.github.petitlapin.Li-ri.appdata.xml#L3 - it should not end in `.desktop`
2. the launchable doesn't match https://github.com/petitlapin/Li-Ri/blob/v3.1.3/io.github.petitlapin.Li-ri.appdata.xml#L18 - it doesn't match the desktop file name.

2 is likely why.

**Comment by petitlapin** [comment]
> 1. the app-id in the appstream is incorrect https://github.com/petitlapin/Li-Ri/blob/v3.1.3/io.github.petitlapin.Li-ri.appdata.xml#L3 - it should not end in `.desktop`
> 
>     2. the launchable doesn't match https://github.com/petitlapin/Li-Ri/blob/v3.1.3/io.github.petitlapin.Li-ri.appdata.xml#L18 - it doesn't match the desktop file name.
> 
> 
> 2 is likely why.

Thank you! I know it was a dumb error but couldn't find it... It should be better now

**Comment by bbhtt** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by bbhtt** [comment]
When the cursor is inside the game window, there seems to be no way no click the window control buttons.

**Comment by petitlapin** [comment]
> When the cursor is inside the game window, there seems to be no way no click the window control buttons like close. Other than that seems fine

On my computer, when I'm in windowed mode, the cursor is changed to the wand (except during play where it's not visible) but if I go outside the window, I have the normal cursor and I can click on the quit button.

**Comment by barthalion** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by barthalion** [comment]
The missing screenshots error comes from the fact you've used URLs to HTML pages instead of actual screenshots, e.g.  https://github.com/petitlapin/Li-Ri/blob/287f0acce15a184ca9e653b4a398b1d2fdb84fd1/fastlane/metadata/android/en-US/images/tenInchScreenshots/01.png

**Comment by petitlapin** [comment]
> The missing screenshots error comes from the fact you've used URLs to HTML pages instead of actual screenshots, e.g. https://github.com/petitlapin/Li-Ri/blob/287f0acce15a184ca9e653b4a398b1d2fdb84fd1/fastlane/metadata/android/en-US/images/tenInchScreenshots/01.png

Thanks, I've replaced it with the [raw links](https://raw.githubusercontent.com/petitlapin/Li-Ri/287f0acce15a184ca9e653b4a398b1d2fdb84fd1/fastlane/metadata/android/en-US/images/tenInchScreenshots/03.png) ([commit](https://github.com/petitlapin/Li-Ri/commit/d20fa514f266311af3dc6ee257f8136b87bca29a)) and fix the summary length.

**Comment by barthalion** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by petitlapin** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by hfiguiere** [comment]
Cmake is trying to git clone.

Build MUST be done offline.

```
1/9] Creating directories for 'simpleini-populate'
[1/9] Performing download step (git clone) for 'simpleini-populate'
Cloning into 'simpleini-src'...
fatal: unable to access 'https://github.com/brofield/simpleini.git/': Could not resolve host: github.com
Cloning into 'simpleini-src'...
fatal: unable to access 'https://github.com/brofield/simpleini.git/': Could not resolve host: github.com
Cloning into 'simpleini-src'...
fatal: unable to access 'https://github.com/brofield/simpleini.git/': Could not resolve host: github.com
-- Had to git clone more than once: 3 times.
```

**Comment by petitlapin** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by petitlapin** [comment]
bot, build io.github.petitlapin.Li-ri

**Comment by petitlapin** [comment]
Thank you for your support, time and patience :)

**Pull request closed without merging**